### DISCUSSION TITLE
Is there a theme editor available anywhere??

### DISCUSSION THREAD

**Discussion by DerellLicht** [dnGrep/dnGrep#1023] [Q&A] [answered]
I would like to make a custom theme for my page, maybe model the first one on the Deviant Dark color theme from CubicExplorer and others... however, hand-editing a 300+ line xaml file is going to be a *lot* of work!!  Is it possible that anyone has developed a GUI editor for dnGrep ??

**Comment by doug24** [comment]
I did create a simple tool to help me with this when I created the themes in dnGrep, and haven't used it since.  It's very rough, wasn't intended to be public, and I never even pushed it to github.  But if you are interested, I could push it. I would have to do some updates to match the current version of the controls. You will need to build it locally to run it - it's all c#, like dnGrep.  

It runs on one theme at a time, and you have to change the selected theme in the code.  You would start with an existing theme file and modify from there.  There is also a color shift tool I that I used to create the Sunset theme from the Dark theme by shifting the blue tones to red/orange.

The left side of the screen is a set of controls to visualize the changes. Not sure if all the controls and control states are represented.  The middle panel is the current theme, and you use the radio buttons to select which attribute color to modify.  The right panel has a bunch of different controls for selecting and adjusting the colors.

![image](https://github.com/dnGrep/dnGrep/assets/17227248/8cadd3af-3e75-4196-92f2-e1cbc1850b05)

**Reply by DerellLicht** [reply]
I'm more experienced with C++ than C#, but I'm willing to take a look at it!!  I have a Github account (Derell Licht), so will be able to access it.  Did you build it with Visual Studio??  If so, will the free version work?

**Comment by doug24** [comment]
Yes, built with Visual Studio 2022, and the free version will work.  

It may be a few days before it's ready, and I'll post back here when it is.

**Reply by DerellLicht** [reply]
Thank ye very much!!!

**Comment by doug24** [comment, marked as answer]
@DerellLicht - it didn't take long, here it is: [Theme Editor](https://github.com/doug24/ThemeEditor)

**Reply by DerellLicht** [reply]
Thank you so much, sire!!  I already have VS2022 installed, I'll begin working on it!!
